__tests__/commands/install/integration.js

```diff
@@ -124,6 +124,25 @@ test.concurrent(
   },
 );
 
+test.concurrent(
+  'resolves the symlinks of other symlinked packages relative to the package using the link: protocol',
+  async () => {
+    await runInstall({}, 'install-link-nested', async (config): Promise<void> => {
+      const expectPath = path.join(config.cwd, 'node_modules', 'b');
```

Comment:
I am worried that we touch what is inside a, probably it should not participate in deduping and be like

```
node_modules/
  a (symlink) -> ../a

a/ 
  b/
  node_modules/
  b (symlink) -> ../b
```